client/tools/settings.coffee

```diff
@@ -81,3 +85,44 @@ defineTool
   click: ->
     return unless (room = currentRoom.get())?
     room.gridSnap.set not room.gridSnap.get()
+
+defineTool
+  name: 'Transparency'
+  category: 'color'
+  icon: 'highlighter'
+  help: 'Change transparency of pen'
+  active: ->
+    allowTransparency.get()
+  click: ->
+    allowTransparency.set not allowTransparency.get()
+    updateOpacity 1.0, allowTransparency.curValue
+  init: ->
+    updateOpacity 1.0, false
+
+
+# These values are chosen for no particular reason.  I saw that
+# 12.5 was a number you liked for highlighting Perhaps .25 should be 12.5
+for opacity in [.75, .50, .25]
+  do (opacity) ->
+    defineTool
+      name: "Opacity:#{opacity*100}"
+      category: 'color'
+      icon: "opacity#{opacity*100}"
+      help: "Select #{opacity*100}% Transparency"
+      click: ->
+        updateOpacity opacity, true
+      active: ->
+        if currentOpacity.get() == opacity then true else false
+
+
+updateOpacity = (val, display) ->
+  allowTransparency.set display
+
+  if currentOpacity.get() == val
+    currentOpacity.set 1.0
+  else
+    currentOpacity.set val
+
+  buttons = document.querySelectorAll('[data-tool^="Opacity"]')
+  for button in buttons
+    button.style.display = if display then "block" else "none"
```

Comment:
Since the move to React, we don't want to manipulate widget DOM directly like this. Instead, add to DrawApp something like:
```coffee
transparency = useTracker -> allowTransparency.get()
...
{if transparency
  <div id="opacities" className="subpalette">
    <ToolCategory category="opacity" placemen="top"/>
  </div>
}
```

Reply:
thanks for making that easier for me